python/pyplugin_installer/installer_data.py

```diff
@@ -397,7 +397,11 @@ def xmlDownloaded(self):
                     fileName = pluginNodes.item(i).firstChildElement("file_name").text().strip()
                     if not fileName:
                         fileName = QFileInfo(pluginNodes.item(i).firstChildElement("download_url").text().strip().split("?")[0]).fileName()
-                    name = fileName.partition(".")[0]
+                    match = re.match('(.*?)[\.-]', fileName)
```

Comment:
It would be nice to use QFileInfo or python os.path methods here

Reply:
@nyalldawson Probably there is a misunderstanding. I'm extracting a part of the base name here, not splitting the file name to the base name and extension. I used QFileInfo's method just a line above, where it was applicable.

Reply:
Ok. I think there's an issue with the regex here though - it need the "r" prefix or the \. won't work as intended

Reply:
@nyalldawson @m-kuhn Could you please explain why and in what conditions it may not work? Honestly I can't see any connection... You mean the former dot or the latter? Of course the backslash is not necessary, I just used to put it for readability, in order to express it's a raw dot.

Reply:
The `\` is currently not escaped (swallowed away by python and not sent to the regexp).

As a test in python

```
print('\.')
print(r'\.')
```

Reply:
Ok, I removed the backslash as it was more confusing than helping... Is it now satisfying for you? ;-) Btw. if you really insist for the raw string, now I can use it, as now there is no risk it would interfere with that backslash ;)

Reply:
Thanks. Python throws warnings because of these, and the warnings appear in the qgis log window. It's not a nice experience for users when a default install raises python warnings by itself...!

Reply:
Ah now I see! I didn't notice those deprecation warnings - there are always tons of them (at least in Debian). Thanks!

Reply:
Thank you @borysiasty 
I realized, that my example actually didn't even proof the point I was trying to make :D